src/Sylius/Bundle/ThemeBundle/Translation/Translator.php

```diff
@@ -121,7 +112,38 @@ protected function initializeCatalogue($locale)
      */
     protected function computeFallbackLocales($locale)
     {
-        return $this->fallbackLocalesProvider->computeFallbackLocales($locale, $this->getFallbackLocales());
+        list($localeWithoutSuffix, $themeSuffix) = $this->getLocaleAndThemeSuffix($locale);
+
+        $computedFallbackLocales = parent::computeFallbackLocales($locale);
+        array_unshift($computedFallbackLocales, $localeWithoutSuffix);
+
+        $fallbackLocales = [];
+        foreach(array_diff($computedFallbackLocales, [$locale]) as $fallbackLocale) {
+            $fallbackWithSuffix = $fallbackLocale . $themeSuffix;
+            if (null !== $themeSuffix && $locale !== $fallbackWithSuffix) {
+                $fallbackLocales[] = $fallbackWithSuffix;
+            }
+
+            $fallbackLocales[] = $fallbackLocale;
+        }
+
+        return array_unique($fallbackLocales);
+    }
+
+    /**
+     * @param string $locale
+     *
+     * @return string[]
+     */
+    private function getLocaleAndThemeSuffix($locale)
```

Comment:
Can we extract it to `getLocaleWithoutSuffix` (suffix -> modifier, wdyt?) and `getLocaleSuffix` (suffix -> modifier, wdyt?)? Or just a simple, plain VO that gives us accessors for locale elements `[language[_territory][.codeset][@modifier]]` - `Locale::fromString('en_US.UTF-8@theme=heron')->getModifier()`?